ibc-core/ics02-client/types/src/height.rs

```diff
@@ -39,9 +39,9 @@ pub struct Height {
 }
 
 impl Height {
-    pub fn new(revision_number: u64, revision_height: u64) -> Result<Self, ClientError> {
+    pub fn new(revision_number: u64, revision_height: u64) -> Result<Self, HeightError> {
```

Comment:
Ah, I just came across this `HeightError`. Doesn't it make more sense to merge it in other existing error enums like `ClientError` and `IdentifierError`? I'm saying that, as per our convention, we've been aiming to have a single module-specific enum wherever possible.

Reply:
> Doesn't it make more sense to merge it in other existing error enums like ClientError and IdentifierError

It's not totally clear to me then when we should return a `ClientError` vs an `IdentifierError` for a `Height`: is a `Height` considered an identifier?

For example, the `Height::new` function returning a `ClientError` doesn't make a lot of sense to me, but perhaps I am missing the rationale?